### ISSUE TITLE
use WithLocalizedMessage unobtrusive issue

### ISSUE THREAD

**Issue by tomsong2016** [JeremySkinner/FluentValidation#397]
Model

 [Validator(typeof(AddUserValidator))]
    public class AddSystemUserModel
    {
        public AddSystemUserModel()
        {
            Id = Guid.NewGuid().ToString();
        }
        public string Id { get; set; }

        [LocalDisplayName("add user name", ResourceName = "AddUserNameDisplayName")]
        public string AddUserName { get; set; }

        private class AddUserValidator : AbstractValidator<AddSystemUserModel>
        {
            /// <summary>
            /// Initializes a new instance of the <see cref="AddUserValidator"/> class.
            /// </summary>
            public AddUserValidator()
            {
                string message = "Name";
                //this.RuleFor(r => r.AddUserName).NotEmpty().WithLocalizedMessage(() => Language.NotInput, x => message);  
            }
        }
}

 public static void NotEmptyWithMessage<T, TProperty>(
            this IRuleBuilder<T, TProperty> ruleBuilder, 
            string propertyChineseName)
        {
            ruleBuilder.NotEmpty().WithLocalizedMessage(()=>Language.NotInput, x => propertyChineseName);           
        }

View 

@using (Ajax.BeginForm("Add", "User", null, ajaxOptions, new { Area = "System", id = "addUserForm", @class = "form-horizontal" }))
        {
        <div class="modal-body">
                <div class="form-group">
                    <label class="col-md-3 col-xs-2 control-label">@Html.GetLangbyKey("AddUserNameDisplayName"):</label>
                    <div class="col-md-9 col-xs-10">
                        @Html.TextBoxFor(a => a.AddUserName, new { @class = "form-control", @placeholder = @Html.GetLangbyKey("AddUserNameDisplayName") })
                        @Html.ValidationMessageFor(a => a.AddUserName, "", new { @class = "text-danger" })
                    </div>
                </div>
   </div>
}

<script src="~/Components/Scripts/jquery.validate.min.js"></script>
<script src="~/Components/Scripts/jquery.validate.unobtrusive.min.js"></script>

<input class="form-control input-validation-error" data-val="true"  **data-val-required="{0} Is required!"** id="AddUserName" name="AddUserName" placeholder="Name" value="" aria-required="true" aria-describedby="AddUserName-error AddUserName-error" aria-invalid="true" type="text">

question:
data-val-required="{0} Is required!“  {0}  is not replaced .

**Comment by JeremySkinner** [comment]
This isn't a fluentvalidation issue. 

Client-side validation is entirely handled by jQuery Validate, which does not support placeholders in this way.

**Comment by tomsong2016** [comment]
Thank you for your reply

But the placeholder is in multi language resources package.

Public static NotEmptyWithMessage<T void (TProperty>)

This IRuleBuilder<T, TProperty> ruleBuilder,

String propertyName)

{

(ruleBuilder.NotEmpty) (.WithLocalizedMessage) (=>Language.NotInput, x = propertyName);

}

The parameters of propertyName to replace the placeholder

Html.TextBoxFor data-val-required read the placeholder should be replaced, can not replace the placeholder? Why?

**Comment by tomsong2016** [comment]
Public static void  NotEmptyWithMessage<T TProperty>(
This IRuleBuilder<T, TProperty> ruleBuilder,
String propertyName)
{
ruleBuilder.NotEmpty() .WithMessage(Language.NotInput.FormatWith(propertyName);
}
Html.TextBoxFor data-val-required read the placeholder should be replaced.

**Comment by JeremySkinner** [comment]
As I said, this isn't a FluentValidation issue.....for clientside validation, it is entirely up to the client technology to format the message. 

FluentValidation only handles placeholders *on the server side*. FluentValidation has nothing to do with how the clientside messages are formatted/displayed.
